### ISSUE TITLE
Don't require pyke

### ISSUE THREAD

**Issue by baerbock** [SciTools/iris#3597]
When installing iris it requires https://github.com/mayask/pyke – which hasn't seen any commit in 10 years and is probably Py2 only.

```
/usr/bin/python -c import os; from pyke.target_pkg import target_pkg; target_pkg.load = lambda *args, **kwargs: None; os.chdir(os.path.join('build/lib', 'iris', 'fileformats', '_pyke_rules')); from pyke import knowledge_engine; knowledge_engine.engine('')
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'pyke'
```

Is there any way to completely disable the tests, or at least those requiring pyke?

**Comment by pp-mo** [comment]
This clearly connects to #3415.

Unfortunately, Pyke is still a _core_ dependency, as we haven't decoupled Iris netcdf decoding from the "core" code.  It's not optional, because it is [imported on the first import of iris](https://github.com/SciTools/iris/blob/738093f1ed639934685725dbc6228c9c9cb8e59f/lib/iris/fileformats/__init__.py#L22).
We should eventually fix this, along with #3415 as above.

In fact, *all* the various fileformat subpackages in `iris.fileformats` are imported before evidence of need, so we can construct a filetype-to-loader mapping.
However an exception is iris-grib, which is now [import-deferred to make it a an optional dependency].  So it can be done!

If you fancy a go at #3415, please give a poke to @bjlittle, who I believe may have some partially-complete solution in this area.

**Comment by rhattersley** [comment]
> we haven't decoupled Iris netcdf decoding from the "core" code

#3657 is an attempt at a first step along that journey

**Comment by bjlittle** [comment]
@baerbock The `pyke` package is available on `conda-forge` and also on PyPI but is called `scitools-pyke`.

As of #4198, which was merged to `main` on 1 July 2021, `pyke` is now no longer a dependency of `iris`.

This will be available in `iris` 3.1.0, which is scheduled for release around September 2021, see the associated [v3.1.x GitHub Discussion](https://github.com/SciTools/iris/discussions/4072).

I'm now going to close this issue. However, please reopen this issue if you want to discuss this further, thanks :+1:
